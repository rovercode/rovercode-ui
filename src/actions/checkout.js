// actions https://redux.js.org/basics/actions
// async actions https://redux.js.org/advanced/asyncactions

import axios from 'axios';

export const CREATE_CHECKOUT_SESSION = 'CREATE_CHECKOUT_SESSION';
export const CREATE_CHECKOUT_SESSION_PENDING = `${CREATE_CHECKOUT_SESSION}_PENDING`;
export const CREATE_CHECKOUT_SESSION_FULFILLED = `${CREATE_CHECKOUT_SESSION}_FULFILLED`;
export const CREATE_CHECKOUT_SESSION_REJECTED = `${CREATE_CHECKOUT_SESSION}_REJECTED`;

export const createCheckoutSession = (userId, lineItems, successURL, cancelURL, collectShippingAddress, xhrOptions) => ({
  type: CREATE_CHECKOUT_SESSION,
  payload: axios.post(`${SUBSCRIPTION_SERVICE}/api/v1/checkout/start/`, { // eslint-disable-line no-undef
    'id': userId,
    'line_items': lineItems,
    'success_url': successURL,
    'cancel_url': cancelURL,
    'collect_shipping_address': collectShippingAddress,
  }, xhrOptions)
    .then(({ data }) => (
      data
    )),
});