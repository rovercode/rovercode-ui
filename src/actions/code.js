// actions https://redux.js.org/basics/actions
// async actions https://redux.js.org/advanced/asyncactions

import axios from 'axios';

export const FETCH_PROGRAM = 'FETCH_PROGRAM';
export const FETCH_PROGRAM_PENDING = `${FETCH_PROGRAM}_PENDING`;
export const FETCH_PROGRAM_FULFILLED = `${FETCH_PROGRAM}_FULFILLED`;
export const FETCH_PROGRAM_REJECTED = `${FETCH_PROGRAM}_REJECTED`;
export const SAVE_PROGRAM = 'SAVE_PROGRAM';
export const SAVE_PROGRAM_PENDING = `${SAVE_PROGRAM}_PENDING`;
export const SAVE_PROGRAM_FULFILLED = `${SAVE_PROGRAM}_FULFILLED`;
export const SAVE_PROGRAM_REJECTED = `${SAVE_PROGRAM}_REJECTED`;
export const CREATE_PROGRAM = 'CREATE_PROGRAM';
export const CREATE_PROGRAM_PENDING = `${CREATE_PROGRAM}_PENDING`;
export const CREATE_PROGRAM_FULFILLED = `${CREATE_PROGRAM}_FULFILLED`;
export const CREATE_PROGRAM_REJECTED = `${CREATE_PROGRAM}_REJECTED`;

export const UPDATE_JSCODE = 'UPDATE_JSCODE';
export const UPDATE_XMLCODE = 'UPDATE_XMLCODE';
export const CHANGE_EXECUTION_STATE = 'CHANGE_EXECUTION_STATE';
export const CHANGE_NAME = 'CHANGE_NAME';
export const CHANGE_NAME_PENDING = `${CHANGE_NAME}_PENDING`;
export const CHANGE_NAME_FULFILLED = `${CHANGE_NAME}_FULFILLED`;
export const CHANGE_NAME_REJECTED = `${CHANGE_NAME}_REJECTED`;
export const CHANGE_PROGRAM_TAGS = 'CHANGE_PROGRAM_TAGS';
export const CHANGE_PROGRAM_TAGS_PENDING = `${CHANGE_PROGRAM_TAGS}_PENDING`;
export const CHANGE_PROGRAM_TAGS_FULFILLED = `${CHANGE_PROGRAM_TAGS}_FULFILLED`;
export const CHANGE_PROGRAM_TAGS_REJECTED = `${CHANGE_PROGRAM_TAGS}_REJECTED`;
export const CHANGE_ID = 'CHANGE_ID';
export const CHANGE_READ_ONLY = 'CHANGE_READ_ONLY';
export const CLEAR_PROGRAM = 'CLEAR_PROGRAM';

// Execution States
export const EXECUTION_RUN = 1;
export const EXECUTION_STEP = 2;
export const EXECUTION_STOP = 3;
export const EXECUTION_RESET = 4;

// action creators
export const updateJsCode = (jsCode) => ({
  type: UPDATE_JSCODE,
  payload: jsCode,
});

export const updateXmlCode = (xmlCode) => ({
  type: UPDATE_XMLCODE,
  payload: xmlCode,
});

export const changeExecutionState = (state) => ({
  type: CHANGE_EXECUTION_STATE,
  payload: state,
});

export const changeName = (id, name, xhroptions) => ({
  type: CHANGE_NAME,
  payload: axios.patch(`/api/v1/block-diagrams/${id}/`, {
    name,
  }, xhroptions)
    .then(({ data }) => (
      data
    )),
});

export const changeId = (id) => ({
  type: CHANGE_ID,
  payload: id,
});

export const fetchProgram = (id, xhroptions) => ({
  type: FETCH_PROGRAM,
  payload: axios.get(`/api/v1/block-diagrams/${id}/`, xhroptions)
    .then(({ data }) => (
      data
    )),
});

export const saveProgram = (id, content, name, lesson, xhroptions) => ({
  type: SAVE_PROGRAM,
  payload: axios.put(`/api/v1/block-diagrams/${id}/`, {
    content,
    name,
    lesson,
  }, xhroptions)
    .then(({ data }) => (
      data
    )),
});

export const createProgram = (name, xhroptions) => ({
  type: CREATE_PROGRAM,
  payload: axios.post('/api/v1/block-diagrams/', {
    name,
    content: '<xml></xml>',
  }, xhroptions)
    .then(({ data }) => (
      data
    )),
});

export const changeProgramTags = (id, tags, xhroptions) => ({
  type: CHANGE_PROGRAM_TAGS,
  payload: axios.patch(`/api/v1/block-diagrams/${id}/`, {
    owner_tags: tags,
  }, xhroptions)
    .then(({ data }) => (
      data
    )),
});

export const changeReadOnly = (isReadOnly) => ({
  type: CHANGE_READ_ONLY,
  payload: isReadOnly,
});

export const clearProgram = () => ({
  type: CLEAR_PROGRAM,
});
