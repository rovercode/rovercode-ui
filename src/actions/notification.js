// actions https://redux.js.org/basics/actions
// async actions https://redux.js.org/advanced/asyncactions

export const SHOW_NOTIFICATION = 'SHOW_NOTIFICATION';
export const CLEAR_NOTIFICATION = 'CLEAR_NOTIFICATION';

// action creators
export const showNotification = (message, duration, severity) => ({
  type: SHOW_NOTIFICATION,
  payload: {
    message,
    duration,
    severity,
  },
});

export const clearNotification = () => ({
  type: CLEAR_NOTIFICATION,
});
