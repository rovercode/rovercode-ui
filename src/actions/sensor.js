// actions https://redux.js.org/basics/actions
// async actions https://redux.js.org/advanced/asyncactions

export const CHANGE_LEFT_SENSOR_STATE = 'CHANGE_LEFT_SENSOR_STATE';
export const CHANGE_RIGHT_SENSOR_STATE = 'CHANGE_RIGHT_SENSOR_STATE';
export const CHANGE_LIGHT_SENSOR_READINGS = 'CHANGE_LIGHT_SENSOR_READINGS';
export const CHANGE_LINE_SENSOR_READINGS = 'CHANGE_LINE_SENSOR_READINGS';
export const CHANGE_DISTANCE_SENSOR_READING = 'CHANGE_DISTANCE_SENSOR_READING';
export const CHANGE_BATTERY_VOLTAGE_READING = 'CHANGE_BATTERY_VOLTAGE_READING';

// Sensor States
export const COVERED = 1;
export const NOT_COVERED = 2;

// action creators
export const changeLeftSensorState = (state) => ({
  type: CHANGE_LEFT_SENSOR_STATE,
  payload: state,
});

export const changeRightSensorState = (state) => ({
  type: CHANGE_RIGHT_SENSOR_STATE,
  payload: state,
});

export const changeLightSensorReadings = (leftReading, rightReading) => ({
  type: CHANGE_LIGHT_SENSOR_READINGS,
  payload: { leftReading, rightReading },
});

export const changeLineSensorReadings = (leftReading, rightReading) => ({
  type: CHANGE_LINE_SENSOR_READINGS,
  payload: { leftReading, rightReading },
});

export const changeDistanceSensorReading = (distanceReading) => ({
  type: CHANGE_DISTANCE_SENSOR_READING,
  payload: distanceReading,
});

export const changeBatteryVoltageReading = (voltageReading) => ({
  type: CHANGE_BATTERY_VOLTAGE_READING,
  payload: voltageReading,
});
