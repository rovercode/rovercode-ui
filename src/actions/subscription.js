// actions https://redux.js.org/basics/actions
// async actions https://redux.js.org/advanced/asyncactions

import axios from 'axios';

export const FETCH_SUBSCRIPTION = 'FETCH_SUBSCRIPTION';
export const FETCH_SUBSCRIPTION_PENDING = `${FETCH_SUBSCRIPTION}_PENDING`;
export const FETCH_SUBSCRIPTION_FULFILLED = `${FETCH_SUBSCRIPTION}_FULFILLED`;
export const FETCH_SUBSCRIPTION_REJECTED = `${FETCH_SUBSCRIPTION}_REJECTED`;
export const UPGRADE_SUBSCRIPTION = 'UPGRADE_SUBSCRIPTION';
export const UPGRADE_SUBSCRIPTION_PENDING = `${UPGRADE_SUBSCRIPTION}_PENDING`;
export const UPGRADE_SUBSCRIPTION_FULFILLED = `${UPGRADE_SUBSCRIPTION}_FULFILLED`;
export const UPGRADE_SUBSCRIPTION_REJECTED = `${UPGRADE_SUBSCRIPTION}_REJECTED`;

// action creators
export const fetchSubscription = (id, xhrOptions) => ({
  type: FETCH_SUBSCRIPTION,
  payload: axios.get(`${SUBSCRIPTION_SERVICE}/api/v1/customer/${id}/`, xhrOptions) // eslint-disable-line no-undef
    .then(({ data }) => (
      data
    )),
});

export const upgradeSubscription = (id, accessCode, xhrOptions) => ({
  type: UPGRADE_SUBSCRIPTION,
  payload: axios.put(`${SUBSCRIPTION_SERVICE}/api/v1/customer/${id}/`, { // eslint-disable-line no-undef
    accessCode,
  }, xhrOptions)
    .then(({ data }) => (
      data
    )),
});
