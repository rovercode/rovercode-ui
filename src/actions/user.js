// actions https://redux.js.org/basics/actions
// async actions https://redux.js.org/advanced/asyncactions
import axios from 'axios';
import jwtDecode from 'jwt-decode';

export const UPDATE_USER = 'UPDATE_USER';
export const EDIT_USER_USERNAME = 'EDIT_USER_USERNAME';
export const EDIT_USER_USERNAME_PENDING = `${EDIT_USER_USERNAME}_PENDING`;
export const EDIT_USER_USERNAME_FULFILLED = `${EDIT_USER_USERNAME}_FULFILLED`;
export const EDIT_USER_USERNAME_REJECTED = `${EDIT_USER_USERNAME}_REJECTED`;
export const EDIT_USER_PASSWORD = 'EDIT_USER_PASSWORD';
export const EDIT_USER_PASSWORD_PENDING = `${EDIT_USER_PASSWORD}_PENDING`;
export const EDIT_USER_PASSWORD_FULFILLED = `${EDIT_USER_PASSWORD}_FULFILLED`;
export const EDIT_USER_PASSWORD_REJECTED = `${EDIT_USER_PASSWORD}_REJECTED`;
export const EDIT_USER_SHOW_GUIDE = 'EDIT_USER_SHOW_GUIDE';
export const EDIT_USER_SHOW_GUIDE_PENDING = `${EDIT_USER_SHOW_GUIDE}_PENDING`;
export const EDIT_USER_SHOW_GUIDE_FULFILLED = `${EDIT_USER_SHOW_GUIDE}_FULFILLED`;
export const EDIT_USER_SHOW_GUIDE_REJECTED = `${EDIT_USER_SHOW_GUIDE}_REJECTED`;
export const FETCH_USER_STATS = 'FETCH_USER_STATS';
export const FETCH_USER_STATS_PENDING = `${FETCH_USER_STATS}_PENDING`;
export const FETCH_USER_STATS_FULFILLED = `${FETCH_USER_STATS}_FULFILLED`;
export const FETCH_USER_STATS_REJECTED = `${FETCH_USER_STATS}_REJECTED`;
export const REFRESH_SESSION = 'REFRESH_SESSION';
export const REFRESH_SESSION_PENDING = `${REFRESH_SESSION}_PENDING`;
export const REFRESH_SESSION_FULFILLED = `${REFRESH_SESSION}_FULFILLED`;
export const REFRESH_SESSION_REJECTED = `${REFRESH_SESSION}_REJECTED`;

// action creators
export const updateUser = (data) => ({
  type: UPDATE_USER,
  payload: data,
});

export const refreshSession = (cookies) => ({
  type: REFRESH_SESSION,
  payload: axios.post('/api/api-token-refresh/', {
    token: cookies.get('auth_jwt'),
  })
    .then(({ data }) => (
      jwtDecode(data.token)
    )),
});

export const editUserUsername = (username, xhrOptions) => ({
  type: EDIT_USER_USERNAME,
  payload: axios.put('/jwt/auth/user/', { username }, xhrOptions)
    .then(({ data }) => (
      data
    )),
});

export const editUserPassword = (password, xhrOptions) => ({
  type: EDIT_USER_PASSWORD,
  payload: axios.post('/jwt/auth/password/change/', {
    new_password1: password,
    new_password2: password,
  }, xhrOptions)
    .then(({ data }) => (
      data
    )),
});

export const editUserShowGuide = (id, show, xhrOptions) => ({
  type: EDIT_USER_SHOW_GUIDE,
  payload: axios.patch(`/api/v1/users/${id}/`, { show_guide: show }, xhrOptions)
    .then(({ data }) => (
      data
    )),
});

export const fetchUserStats = (id, xhroptions) => ({
  type: FETCH_USER_STATS,
  payload: axios.get(`/api/v1/users/${id}/stats/`, xhroptions)
    .then(({ data }) => (
      data
    )),
});
