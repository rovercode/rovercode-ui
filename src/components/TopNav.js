import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link, Redirect } from 'react-router-dom';
import {
  withStyles,
} from '@material-ui/core/styles';
import MenuItem from '@material-ui/core/MenuItem';
import Menu from '@material-ui/core/Menu';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
import Grid from '@material-ui/core/Grid';
import { Box } from '@material-ui/core';
import { FormattedMessage } from 'react-intl';
import { withCookies, Cookies } from 'react-cookie';
import { logout as actionLogout } from '@/actions/auth';

const mapDispatchToProps = (dispatch) => ({
  logout: () => dispatch(actionLogout()),
});

class TopNav extends Component {
  constructor(props) {
    super(props);

    this.state = {
      redirectToLogin: false,
      userMenuAnchorElement: null,
    };
  }

  signout = () => {
    const { cookies, logout } = this.props;

    cookies.remove('auth_jwt', { path: '/' });
    logout();

    this.setState({ redirectToLogin: true });
  }

  handleMenuOpen = (event) => {
    this.setState({
      userMenuAnchorElement: event.target,
    });
  };

  handleMenuClose = () => {
    this.setState({
      userMenuAnchorElement: null,
    });
  };

  render() {
    const { userName } = this.props;
    const { redirectToLogin, userMenuAnchorElement } = this.state;

    const NavBarSpacer = withStyles((theme) => ({
      root: {
        marginBottom: theme.spacing(2),
      },
    }))(Box);

    return (
      <>
        {
          redirectToLogin ? (
            <Redirect to="/accounts/login" />
          ) : (null)
        }

        <div>
          <AppBar position="static" color="secondary">
            <Toolbar>
              <Grid item container direction="row" spacing={4}>
                <Grid item xs={2}>
                  <Box
                    display="flex"
                    justifyContent="left"
                  >
                    <Button
                      size="large"
                      variant="contained"
                      disableElevation
                      color="secondary"
                      component={Link}
                      to="/"
                    >
                      <Typography variant="h6">
                        Logo/Connect
                      </Typography>
                    </Button>
                  </Box>
                </Grid>
                <Grid item xs={2}>
                  <Box
                    display="flex"
                    justifyContent="center"
                  >
                    <Button
                      size="large"
                      variant="contained"
                      disableElevation
                      color="secondary"
                      component={Link}
                      to="/programs"
                    >
                      <Typography variant="h6">
                        <FormattedMessage
                          id="app.top_nav.my_programs"
                          description="Button label to go to the user's own programs"
                          defaultMessage="My Programs"
                        />
                      </Typography>
                    </Button>
                  </Box>
                </Grid>
                <Grid item xs={2}>
                  <Box
                    display="flex"
                    justifyContent="center"
                  >
                    <Button
                      size="large"
                      variant="contained"
                      disableElevation
                      color="secondary"
                      component={Link}
                      to="/programs"
                    >
                      <Typography variant="h6">
                        <FormattedMessage
                          id="app.top_nav.community_programs"
                          description="Button label to go to the community programs"
                          defaultMessage="Community Programs"
                        />
                      </Typography>
                    </Button>
                  </Box>
                </Grid>
                <Grid item xs={2}>
                  <Box
                    display="flex"
                    justifyContent="center"
                  >
                    <Button
                      size="large"
                      variant="contained"
                      disableElevation
                      color="secondary"
                      component={Link}
                      to="/programs"
                    >
                      <Typography variant="h6">
                        <FormattedMessage
                          id="app.top_nav.courses"
                          description="Button label to go to the courses"
                          defaultMessage="Courses"
                        />
                      </Typography>
                    </Button>
                  </Box>
                </Grid>
              </Grid>
              <Button
                size="large"
                variant="contained"
                disableElevation
                color="secondary"
                endIcon={<ArrowDropDownIcon />}
                aria-label="account of current user"
                aria-controls="user-menu"
                aria-haspopup="true"
                onClick={this.handleMenuOpen}
              >
                <Typography variant="h6">
                  {userName}
                </Typography>
              </Button>
              <Menu
                id="user-menu"
                anchorEl={userMenuAnchorElement}
                keepMounted
                open={Boolean(userMenuAnchorElement)}
                onClose={this.handleMenuClose}
                getContentAnchorEl={null}
                anchorOrigin={{
                  vertical: 'bottom',
                  horizontal: 'center',
                }}
                transformOrigin={{
                  vertical: 'top',
                  horizontal: 'center',
                }}
              >
                <MenuItem component={Link} to="/user/settings">
                  <FormattedMessage
                    id="app.top_nav.settings"
                    description="Button label to go to settings"
                    defaultMessage="Settings"
                  />
                </MenuItem>
                <MenuItem onClick={this.signout}>
                  <FormattedMessage
                    id="app.top_nav.sign_out"
                    description="Button label to sign out"
                    defaultMessage="Sign Out"
                  />
                </MenuItem>
              </Menu>
            </Toolbar>
          </AppBar>
          <NavBarSpacer />
        </div>
      </>
    );
  }
}

TopNav.defaultProps = {
  userName: '',
};

TopNav.propTypes = {
  cookies: PropTypes.instanceOf(Cookies).isRequired,
  userName: PropTypes.string,
  logout: PropTypes.func.isRequired,
};

export default withCookies(connect(null, mapDispatchToProps)(TopNav));
