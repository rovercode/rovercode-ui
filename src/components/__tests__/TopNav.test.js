import React from 'react';
import { Redirect } from 'react-router-dom';
import { shallow } from 'enzyme';
import { Cookies } from 'react-cookie';
import configureStore from 'redux-mock-store';
import { logout } from '@/actions/auth';
import TopNav from '../TopNav';

const cookies = new Cookies();
const mockStore = configureStore();
const store = mockStore();
store.dispatch = jest.fn();

describe('The TopNav component', () => {
  beforeEach(() => {
    cookies.set('auth_jwt', '1234', { path: '/' });
  });

  test('should render on the page with no errors', () => {
    const topNav = shallow(<TopNav userName="Dale Gribble" store={store} />, {
      context: { cookies },
    });
    const wrapper = topNav.dive().dive().dive().dive();

    expect(wrapper).toMatchSnapshot();
  });

  test('should sign out', () => {
    const topNav = shallow(<TopNav userName="Dale Gribble" store={store} />, {
      context: { cookies },
    });
    const wrapper = topNav.dive().dive().dive().dive();

    expect(wrapper.find(Redirect).exists()).toBe(false);
    expect(cookies.get('auth_jwt', { path: '/' })).toBe('1234');

    wrapper.instance().signout();
    wrapper.update();

    expect(wrapper.find(Redirect).exists()).toBe(true);
    expect(wrapper.find(Redirect).prop('to')).toBe('/accounts/login');
    expect(cookies.get('auth_jwt', { path: '/' })).toBeUndefined();
    expect(store.dispatch).toHaveBeenCalledWith(logout());
  });

  test('should set and clear menu anchor element when menu is opening and closing', () => {
    const topNav = shallow(<TopNav userName="Dale Gribble" store={store} />, {
      context: { cookies },
    });
    const wrapper = topNav.dive().dive().dive().dive();

    expect(wrapper.instance().state.userMenuAnchorElement).toBe(null);
    wrapper.instance().handleMenuOpen({ target: 'element' });
    expect(wrapper.instance().state.userMenuAnchorElement).toBe('element');
    wrapper.instance().handleMenuClose();
    expect(wrapper.instance().state.userMenuAnchorElement).toBe(null);
  });
});
