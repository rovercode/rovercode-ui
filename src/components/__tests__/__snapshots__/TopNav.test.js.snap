// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The TopNav component should render on the page with no errors 1`] = `
<Fragment>
  <div>
    <WithStyles(ForwardRef(AppBar))
      color="secondary"
      position="static"
    >
      <WithStyles(ForwardRef(Toolbar))>
        <WithStyles(ForwardRef(Grid))
          container={true}
          direction="row"
          item={true}
          spacing={4}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={2}
          >
            <Styled(MuiBox)
              display="flex"
              justifyContent="left"
            >
              <WithStyles(ForwardRef(Button))
                color="secondary"
                component={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "Link",
                    "propTypes": Object {
                      "innerRef": [Function],
                      "onClick": [Function],
                      "replace": [Function],
                      "target": [Function],
                      "to": [Function],
                    },
                    "render": [Function],
                  }
                }
                disableElevation={true}
                size="large"
                to="/"
                variant="contained"
              >
                <WithStyles(ForwardRef(Typography))
                  variant="h6"
                >
                  Logo/Connect
                </WithStyles(ForwardRef(Typography))>
              </WithStyles(ForwardRef(Button))>
            </Styled(MuiBox)>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={2}
          >
            <Styled(MuiBox)
              display="flex"
              justifyContent="center"
            >
              <WithStyles(ForwardRef(Button))
                color="secondary"
                component={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "Link",
                    "propTypes": Object {
                      "innerRef": [Function],
                      "onClick": [Function],
                      "replace": [Function],
                      "target": [Function],
                      "to": [Function],
                    },
                    "render": [Function],
                  }
                }
                disableElevation={true}
                size="large"
                to="/programs"
                variant="contained"
              >
                <WithStyles(ForwardRef(Typography))
                  variant="h6"
                >
                  <FormattedMessage
                    defaultMessage="My Programs"
                    id="app.top_nav.my_programs"
                    values={Object {}}
                  />
                </WithStyles(ForwardRef(Typography))>
              </WithStyles(ForwardRef(Button))>
            </Styled(MuiBox)>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={2}
          >
            <Styled(MuiBox)
              display="flex"
              justifyContent="center"
            >
              <WithStyles(ForwardRef(Button))
                color="secondary"
                component={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "Link",
                    "propTypes": Object {
                      "innerRef": [Function],
                      "onClick": [Function],
                      "replace": [Function],
                      "target": [Function],
                      "to": [Function],
                    },
                    "render": [Function],
                  }
                }
                disableElevation={true}
                size="large"
                to="/programs"
                variant="contained"
              >
                <WithStyles(ForwardRef(Typography))
                  variant="h6"
                >
                  <FormattedMessage
                    defaultMessage="Community Programs"
                    id="app.top_nav.community_programs"
                    values={Object {}}
                  />
                </WithStyles(ForwardRef(Typography))>
              </WithStyles(ForwardRef(Button))>
            </Styled(MuiBox)>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={2}
          >
            <Styled(MuiBox)
              display="flex"
              justifyContent="center"
            >
              <WithStyles(ForwardRef(Button))
                color="secondary"
                component={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "Link",
                    "propTypes": Object {
                      "innerRef": [Function],
                      "onClick": [Function],
                      "replace": [Function],
                      "target": [Function],
                      "to": [Function],
                    },
                    "render": [Function],
                  }
                }
                disableElevation={true}
                size="large"
                to="/programs"
                variant="contained"
              >
                <WithStyles(ForwardRef(Typography))
                  variant="h6"
                >
                  <FormattedMessage
                    defaultMessage="Courses"
                    id="app.top_nav.courses"
                    values={Object {}}
                  />
                </WithStyles(ForwardRef(Typography))>
              </WithStyles(ForwardRef(Button))>
            </Styled(MuiBox)>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Button))
          aria-controls="user-menu"
          aria-haspopup="true"
          aria-label="account of current user"
          color="secondary"
          disableElevation={true}
          endIcon={<Memo(ArrowDropDownIcon) />}
          onClick={[Function]}
          size="large"
          variant="contained"
        >
          <WithStyles(ForwardRef(Typography))
            variant="h6"
          >
            Dale Gribble
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Button))>
        <WithStyles(ForwardRef(Menu))
          anchorEl={null}
          anchorOrigin={
            Object {
              "horizontal": "center",
              "vertical": "bottom",
            }
          }
          getContentAnchorEl={null}
          id="user-menu"
          keepMounted={true}
          onClose={[Function]}
          open={false}
          transformOrigin={
            Object {
              "horizontal": "center",
              "vertical": "top",
            }
          }
        >
          <WithStyles(ForwardRef(MenuItem))
            component={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "Link",
                "propTypes": Object {
                  "innerRef": [Function],
                  "onClick": [Function],
                  "replace": [Function],
                  "target": [Function],
                  "to": [Function],
                },
                "render": [Function],
              }
            }
            to="/user/settings"
          >
            <FormattedMessage
              defaultMessage="Settings"
              id="app.top_nav.settings"
              values={Object {}}
            />
          </WithStyles(ForwardRef(MenuItem))>
          <WithStyles(ForwardRef(MenuItem))
            onClick={[Function]}
          >
            <FormattedMessage
              defaultMessage="Sign Out"
              id="app.top_nav.sign_out"
              values={Object {}}
            />
          </WithStyles(ForwardRef(MenuItem))>
        </WithStyles(ForwardRef(Menu))>
      </WithStyles(ForwardRef(Toolbar))>
    </WithStyles(ForwardRef(AppBar))>
    <WithStyles(Styled(MuiBox)) />
  </div>
</Fragment>
`;
